File: projects/JavaFXApps/core/src/main/java/net/rickiekarp/core/ui/windowmanager/WindowContentController.java
package net.rickiekarp.core.ui.windowmanager;

import net.rickiekarp.core.components.button.SidebarButton;
import javafx.scene.layout.HBox;
import javafx.scene.layout.VBox;

import java.util.ArrayList;
import java.util.List;

public class WindowContentController {
    private List<SidebarButton> sidebarNodes;
    private HBox titlebarRightButtonBox;
    private VBox sidebarButtonBox;

    WindowContentController() {
        sidebarNodes = new ArrayList<>(3);
    }

    public void addSidebarItem(int position, SidebarButton node) {
        sidebarNodes.add(position, node);
    }

    void addSidebarItem(SidebarButton node) {
        addSidebarItem(sidebarNodes.size(), node);
    }

    /**
     * Removes a SidebarButton item by its given identifier
     * @param identifier SidebarButton identifier
     */
    public void removeSidebarItemByIdentifier(String identifier) {
        for (int i = 0; i < sidebarNodes.size(); i++) {
            if (sidebarNodes.get(i).getIdentifier().equals(identifier)) {
                sidebarNodes.remove(i);
            }
        }
        sidebarButtonBox.getChildren().setAll(sidebarNodes);
    }

    List<SidebarButton> getList() {
        return sidebarNodes;
    }

    HBox getTitlebarButtonBox() {
        return titlebarRightButtonBox;
    }

    VBox getSidebarButtonBox() {
        return sidebarButtonBox;
    }

    void setTitlebarRightButtonBox(HBox titlebar) {
        this.titlebarRightButtonBox = titlebar;
    }

    void setSidebarButtonBox(VBox sidebar) {
        this.sidebarButtonBox = sidebar;
    }
}

File: projects/JavaFXApps/core/src/main/java/net/rickiekarp/core/ui/windowmanager/WindowContentController.kt
package net.rickiekarp.core.ui.windowmanager

import net.rickiekarp.core.components.button.SidebarButton
import javafx.scene.layout.HBox
import javafx.scene.layout.VBox

import java.util.ArrayList

class WindowContentController internal constructor() {
    private val sidebarNodes: MutableList<SidebarButton>
    var titlebarButtonBox: HBox? = null
        private set
    var sidebarButtonBox: VBox? = null

    val list: List<SidebarButton>
        get() = sidebarNodes

    init {
        sidebarNodes = ArrayList(3)
    }

    fun addSidebarItem(position: Int, node: SidebarButton) {
        sidebarNodes.add(position, node)
    }

    fun addSidebarItem(node: SidebarButton) {
        addSidebarItem(sidebarNodes.size, node)
    }

    /**
     * Removes a SidebarButton item by its given identifier
     * @param identifier SidebarButton identifier
     */
    fun removeSidebarItemByIdentifier(identifier: String) {
        for (i in sidebarNodes.indices) {
            if (sidebarNodes[i].identifier == identifier) {
                sidebarNodes.removeAt(i)
            }
        }
        sidebarButtonBox!!.children.setAll(sidebarNodes)
    }

    fun setTitlebarRightButtonBox(titlebar: HBox) {
        this.titlebarButtonBox = titlebar
    }
}
